### ISSUE TITLE
🐛 "no entry found for key" panic after moving a bunch of files

### ISSUE THREAD

**Issue by itsMapleLeaf** [biomejs/biome#2758]
### Environment information

```block
CLI:
  Version:                      1.7.2
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         unset
  JS_RUNTIME_VERSION:           "v20.11.1"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "pnpm/9.1.0"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 false

Workspace:
  Open Documents:               0
```

### What happened?

I got this message after moving around a bunch of files and running `biome check .`:

> Biome encountered an unexpected error
> 
> This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue:
> 
> Source Location: crates\biome_js_semantic\src\semantic_model\scope.rs:115:33
> Thread Name: biome::worker_15
> Message: no entry found for key

Restarting the LSP or the editor doesn't seem to help

### Expected result

To check files without raising this error

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Conaclos** [comment]
It is a known issue from Biome 1.7.2.
It was fixed in Biome 1.7.3.
Please upgrade Biome :)
